### PR TITLE
draft of release 1.28 blog post

### PR DESCRIPTION
**Pull request by solvingj**
needs at least some links, potentially json snippets of lockfiles, and some closing remarks.  also some more info about pkg-config components.

Ping @memsharded for review of general focus and content for accuracy

### CODE REVIEW

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+generators. With this release, we add support to the `pkg-config` generator.
+
+## Define generator filenames (CMake Only)
+
+All generators produce the generated files using the naming convnetion of:
+`conanbuildinfo.xyz` where `xyz` is the typical file extension associated with
+the corresonding build system or platform. This was unconfigurable. Over the
+past year, this convention has been revisited for a few different reasons. The
+change for this particular release was to enable recipe authors to set the
+generated build system file name as an attribute on `cpp_info`.
```
Comment (lasote) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:138]: What is the attribute?

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+This "base lockfile" concept leads to the next major improvement: stronger
+enforcement of "locked" values making each field in a lockfile "truly
+immutable". This is not to say that a lockfile cannot be
+updated (lockfiles can and are updated). The rule is this: empty fields in a
+lockfile can be populated, but once a field has a value, it cannot change. Base
+lockfiles are a good example of cases which have unpopulated fields which are
+waiting for future commands to set values in. This improvement was very
+important, as it was recently discovered that there were several ways to
+produce completely invalid and unbuildable lockfiles in previous versions. That
+should no longer be possible moving forward.
```
Comment (memsharded) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:61]: Add a link to the section in the docs.

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+at different times. Even when done "in parallel", they can be slightly offset.
+The problem is that other jobs can be uploading new versions or revisions of
+dependencies at any time. Thus, running `conan lock` multiple times in different
+workspaces, even just a few seconds apart, could result in different
+configurations locking different revisions of the same dependency. Now, CI jobs
+can begin by calling `conan lock` with the `--base` flag. This will produce a
+"general" lockfile which only locks the name and versions of dependencies. A
+copy of this lockfile can be passed to `conan create` in each of the future
+build stages that follow, which will add the configuration-specific fields and
+information to the copies.
```
Comment (memsharded) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:50]: Very well explained, good job.

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+will be in violation of the "One-Definition Rule" (ODR) of C++ and receive fatal
+compiler errors.  By adding this attribute to Conan, we can detect the violation
+based on the attribute before compilation begins, saving significant time in a
+number of cases.
+
+`required_conan_version` enables users to declare what minimum version of Conan
+a recipe requires. With the rapid evolution of Conan and new features coming out
+every month, this attribute makes a lot of sense. If you are familiar with the
+`CMake` build system, you might recognize that this is similar to
+`cmake_required_version`.
```
Comment (memsharded) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:81]: Maybe recall here that there is already a conan.conf one, that can be used in enterprises to guarantee client versions for all devs and CIs

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+All generators produce the generated files using the naming convnetion of:
+`conanbuildinfo.xyz` where `xyz` is the typical file extension associated with
+the corresonding build system or platform. This was unconfigurable. Over the
+past year, this convention has been revisited for a few different reasons. The
+change for this particular release was to enable recipe authors to set the
+generated build system file name as an attribute on `cpp_info`.  
+
+## Additional Features and Fixes  
+
+Add content
```
Comment (memsharded) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:142]: Add here a couple of minor things, like the ``init()`` thingy, and maybe a very short sentence about things starting to move for the Conan 2.0, like some scaffolding for the new cli2.0, and it is done.

**Comment by solvingj** [comment]
Ping @SSE4 : would you be willing to proofread the section on Clang/Windows support?

File: _posts/2020-08-06-New-conan-release-1-28.markdown [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+filenames"
+---
+
+Conan 1.28 has been released. With a long list of new features, it definitely
+feels like progress! Lockfiles were basically completely overhauled, so lets
+start there.
+
+## Lockfile Improvements
+
+First, the command itself has complete changed. What was previously:
```
Comment (SSE4) [inline _posts/2020-08-06-New-conan-release-1-28.markdown:15]:
```suggestion
First, the command itself has completely changed. What was previously:
```

**Comment by SSE4** [comment]
looks very good for me.
some question (not related to Clang):
```
This is not to say that a lockfile cannot be
updated (lockfiles can and are updated).
```
if I want lock file to be truly immutable (e.g. to ensure conan never adds new fields or values into lock file), how can I ensure that?

**Comment by solvingj** [comment]
> if I want lock file to be truly immutable (e.g. to ensure conan never adds new fields or values into lock file), how can I do that?

A lockfile becomes immutable when all the fields in it are populated with values.  It cannot be considered "fully reproducible" as long as it has unpopulated values, and therefor we do not make it immutable.  It's can be thought of like an object which has only been partially constructed, and therefor not fully valid for it's primary use. 

When starting with a "base" lockfile, you'll generally want to use it repeatedly to produce lockfiles for different configurations. You won't want to modify it, so in that case you specify `--lockfile-out` and it will produce a new lockfile and leave the base untouched.  But, that's a matter of process and convention, not guaranteed by Conan.

File: _posts/2020-08-06-New-conan-release-1-28.markdown (hunk 1 above)
Comment (solvingj) [reply _posts/2020-08-06-New-conan-release-1-28.markdown]: Thanks! Resolved!

File: _posts/2020-08-06-New-conan-release-1-28.markdown (hunk 6 above)
Comment (solvingj) [reply _posts/2020-08-06-New-conan-release-1-28.markdown]: fixed in my commit

**Pull request merged**